dataset/batch_image.py

```diff
@@ -1,525 +1,717 @@
+        Actually a wrapper for scipy.misc.imsave.
+
+        .. note:: `components` must be str
 
         Parameters
-        -----------
+        ----------
+        dst : str
+            Folder where to dump. If dst is None then it is determined from index.
         components : str
-            a component name or names which data should be rotated
-
-        angle : float
-            the rotation angle in degrees.
-        preserve_shape : bool
-            whether to keep shape after rotating
-            (always True for images as arrays, can be False for PIL.Images)
+            component to save.
         """
-        return self._rotate_one(ix, components, angle, preserve_shape, **kwargs)
+        _ = self, ix, dst, components
+        raise NotImplementedError("Must be implemented in a child class")
 
     @action
-    @inbatch_parallel(init='indices', post='assemble')
-    def random_rotate(self, ix, components='images', p=1., angle=None, **kwargs):
-        """ Rotate each image in the batch at a random angle
+    def dump(self, dst=None, fmt=None, components="images", *args, **kwargs):
+        """ Dump data.
 
         Parameters
-        -----------
-        components : str
-            a component name or names which data should be rotated
+        ----------
+        dst : str, None
+            Path to the folder where to dump. If dst is None then path is determined from index.
+        fmt : {'image', 'blosc', 'csv', 'hdf5', 'feather'}
+            Format of the file to save.
+        components : str, sequence
+            components to save. Note that if `fmt='images'` than components must be str.
+
+        Returns
+        -------
+        self
+        """
+        if fmt == 'image':
+            self._dump_image(dst, components)
+        return super().dump(dst, fmt, components, *args, **kwargs)
 
-        p : float
-            a probability to apply rotate
-            (0. - don't rotate, .5 - rotate half of images, 1 - rotate all images)
 
-        angle : tuple
-            an angle range in the form of (min_angle, max_angle), in degrees
-        """
-        if np.random.binomial(1, p) > 0:
-            angle = angle or (-45., 45.)
-            _angle = np.random.uniform(*angle)
-            preserve_shape = kwargs.pop('preserve_shape', True)
-            return self._rotate_one(ix, components, _angle, preserve_shape=preserve_shape, **kwargs)
-        image = self.get(ix, components)
-        return image
+@transform_actions(prefix='_', suffix='_all', wrapper='apply_transform_all')
+@transform_actions(prefix='_', suffix='_', wrapper='apply_transform')
+class ImagesBatch(BaseImagesBatch):
+    """ Batch class for 2D images.
 
-    @action
-    def flip(self, axis=None):
-        """ Flip images
+    Images are stored as numpy arrays (N, H, W, C).
+    """
+
+    @property
+    def image_shape(self):
+        """: tuple - shape of the image """
+        return self.images.shape[1:]
```

Comment:
It will give an unexpected result if images in the batch have different shapes.
Check and raise an error where applicable.

Reply:
+